spec/pricing/nsm_spec.rb

```diff
@@ -151,11 +151,155 @@
     end
   end
 
+  describe "#youth_court_fee" do
+    let(:claim) do
+      {
+        claim_type:,
+        rep_order_date: claim_date,
+        vat_registered: true,
+        claimed_youth_court_fee_included:,
+        assessed_youth_court_fee_included:,
+        youth_court:,
+        plea_category:,
+      }
+    end
+    let(:claim_type) { "non_standard_magistrate" }
+    let(:claim_date) { "2024-10-10" }
+    let(:vat_registered) { true }
+    let(:claimed_youth_court_fee_included) { true }
+    let(:assessed_youth_court_fee_included) { nil }
+    let(:youth_court) { true }
+    let(:plea_category) { "category_1a" }
+
+    context "claim date pre december 6th (Non-standard magistrate)" do
+      it "does not apply the fee" do
+        expect(described_class.calculate_youth_court_fee(claim)).to eq({
+          claimed_total_exc_vat: 0.0,
+          assessed_total_exc_vat: 0.0,
+          claimed_vatable: 0.0,
+          assessed_vatable: 0.0,
+        })
+      end
+    end
+
+    context "claim date pre december 6th (Breach of inunction)" do
```

Comment:
There's something wonky about this test - the claim_type isn't checked by the code, and doesn't affect the outcome. The thing that's preventing this fee from being applied is the date, so this whole test is basically just repeating the previous one.

If you wanted to test claim_type, what you'd want is a test where the date is _after_ 6 December and the claim type is breach_of_injunction, and you'd want to verify that that didn't apply the fee - and I think that test would fail, because as I say, you don't actually check the claim_type in the class.

Reply:
good catch - I misinterpretted the requirement (was thinking claim is an nsm claim, not claim type of an nsm claim is non_standard_magistrate as opposed to breach_of_injunction)